spec/rubocop/cop/rails/unique_validation_without_index_spec.rb

```diff
@@ -137,6 +137,53 @@ class WrittenArticles
       end
     end
 
+    context 'when the validation is for a polymorphic association' do
+      context 'without proper index' do
+        let(:schema) { <<~RUBY }
+          ActiveRecord::Schema.define(version: 2020_02_02_075409) do
+            create_table "written_articles", force: :cascade do |t|
+              t.string "title", null: false
+              t.string "author_type", null: false
+              t.bigint "author_id", null: false
+              t.index ["title", "author_id"], name: "idx_aid", unique: true
+            end
+          end
+        RUBY
+
+        it 'registers an offense' do
+          expect_offense(<<~RUBY)
+            class WrittenArticles
+              belongs_to :author, polymorphic: true
+              validates :title, uniqueness: { scope: :author }
+              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Uniqueness validation should be with a unique index.
+            end
+          RUBY
+        end
```

Comment:
It looks like the following test cases need to be passed if my understanding is correct. Can you take a look?

```suggestion
        end

        it 'does not register an offense when `polymorphic: false`' do
          expect_no_offenses(<<~RUBY)
            class WrittenArticles
              belongs_to :author, polymorphic: false
              validates :title, uniqueness: { scope: :author }
            end
          RUBY
        end

        it 'does not register an offense when `polymorphic` option is not specified' do
          expect_no_offenses(<<~RUBY)
            class WrittenArticles
              belongs_to :author
              validates :title, uniqueness: { scope: :author }
            end
          RUBY
        end
```

Reply:
Good suggestion---I wasn't actually checking the model class to see if the relation was polymorphic, but rather just relying on naming conventions. I added the specs you suggested and reworked the logic a bit to use the model class to determine if the relation is polymorphic.